src/Workspaces/CSharp/Portable/Extensions/ContextQuery/SyntaxTreeExtensions.cs

```diff
@@ -2218,6 +2218,17 @@ public static bool IsIsOrAsContext(this SyntaxTree syntaxTree, int position, Syn
             // is/as are valid after expressions.
             if (token.IsLastTokenOfNode<ExpressionSyntax>())
             {
+                // Not after lambdas, anonymous methods, or
+                // expressions of type 'void' or 'method group'
+                var expression = token.GetAncestors<ExpressionSyntax>().LastOrDefault();
+                var leftHandBinding = semanticModel.GetSymbolInfo(expression);
+                var symbol = leftHandBinding.GetBestOrAllSymbols().FirstOrDefault();
+                var memberType = symbol?.GetMemberType();
+                if (memberType != null && memberType.SpecialType == SpecialType.System_Void)
+                {
+                    return false;
+                }
+
```

Comment:
Yeah, good call, it definitely doesn't. I got caught up in the void return from the Action examples in the original issue. This would not handle a Func<int> delegate though.